native/packages/react-native-bpk-component-horizontal-nav/src/BpkHorizontalNav.js

```diff
@@ -35,48 +40,157 @@ const styles = StyleSheet.create({
     borderColor: 'transparent',
     borderBottomColor: tokens.colorGray100,
     borderWidth: tokens.borderSizeSm,
+  },
+  horizontalNavInner: {
     flexDirection: 'row',
   },
   spaceAround: {
     flex: 0,
     flexGrow: 1,
     justifyContent: 'space-around',
   },
+  selectedIndicator: {
+    backgroundColor: tokens.colorBlue700,
+    height: tokens.borderSizeLg,
+  },
 });
 
-const BpkHorizontalNav = (props) => {
-  const {
-    children,
-    spaceAround,
-    style,
-    ...rest
-  } = props;
-  const navStyle = [styles.horizontalNav];
-  if (spaceAround) {
-    navStyle.push(styles.spaceAround);
+const validateSelectedIndex = (selectedIndex, childrenCount) => {
+  if (typeof selectedIndex !== 'number' || !Number.isInteger(selectedIndex)) {
+    return false;
   }
-  return (
-    <ScrollView
-      alwaysBounceHorizontal={false}
-      contentContainerStyle={[navStyle, style]}
-      horizontal
-      showsHorizontalScrollIndicator={false}
-      {...rest}
-    >
-      { children }
-    </ScrollView>
-  );
+  if (selectedIndex < 0 || selectedIndex + 1 > childrenCount) {
+    return false;
+  }
+  return true;
 };
 
-BpkHorizontalNav.propTypes = {
-  children: PropTypes.node.isRequired,
+class BpkHorizontalNav extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      indicatorOffsetX: null,
+      indicatorWidth: null,
+    };
+    this.onChildLayout = this.onChildLayout.bind(this);
+    this.animateIndicator = this.animateIndicator.bind(this);
+  }
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.selected !== null) {
+      this.animateIndicator(this.childrenPositions[nextProps.selected]);
+    }
+  }
+  onChildLayout = (event, index) => {
```

Comment:
We could put the index as the first argument here and then partially apply the function before passing it to the `BpkHorizontalNavItem`. It's slightly nicer than `onLayout: event => this.onChildLayout(event, index)` since it instead could just be `onLayout: boundOnChildLayout`